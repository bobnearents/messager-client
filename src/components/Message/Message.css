p {
  background:#F5E9B3;
  width:30%;
  padding:8px 20px;
  margin:5px;
  border-radius:15px;
}

.mine {
  background:#0072c6;
  align-self: flex-end;
  color:white;
}