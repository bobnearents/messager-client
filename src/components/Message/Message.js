import React, { Component } from 'react';
import './Message.css';

class Message extends Component {
  constructor(props) {
    super(props);
    this.state = { }
  }

  prettifyDate = (ISOdate) => {
    let date = new Date(ISOdate)
    const year = date.getFullYear();
    const month = date.getMonth() + 1;
    const dt = date.getDate();
    const time = date.getTime();

    return(month + '/' + dt);
  }
  render() { 
    return (
      <>
        <p className = {this.props.mine}>
          <big>{this.props.text}</big>
          <br />
          <i>
            <small>
              {this.props.user} 
            </small>
          </i>
        </p>
        <br />
      </>
    );
  }
}
 
export default Message;