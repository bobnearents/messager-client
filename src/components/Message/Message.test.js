import React from 'react';
import { shallow } from 'enzyme';
import renderer from "react-test-renderer";

import Message from './Message';

describe('<Message />', () => {
  
  it('Renders without crashing', () => {
    shallow(<Message />)
  });

  it("renders correctly", () => {
    const tree = renderer.create(<Message />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it('Renders the text', () => {
    const text = "Foo";
    const wrapper = shallow(<Message text={text} />);
    expect(wrapper.text().split(" ")[0]).toEqual(text);
  });

  it("Renders the text", () => {
    const text = "Foo";
    const wrapper = shallow(<Message user={text} />);
    expect(wrapper.text().split(" ")[0]).toEqual(text);
  });



})