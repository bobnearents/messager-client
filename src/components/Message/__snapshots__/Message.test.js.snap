// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> renders correctly 1`] = `
Array [
  <p>
    <big />
    <br />
    <i>
      <small />
    </i>
  </p>,
  <br />,
]
`;
